File: 1103. Distribute Candies to People/Solution.cpp
class Solution {
public:
    vector<int> distributeCandies(int candies, int num_people) {
        vector<int> vCandies(num_people, 0);

        int turn = 1;
        int counter = 0;
        while(candies > 0)
        {
            if (candies >= turn)
            {
                vCandies[counter] += turn;
                candies -= turn;
                counter++;
                turn++;
                counter %= num_people;
                
            } else
            {
                vCandies[counter] += candies;
                break;
            }
        }
        return vCandies;
    }
};

File: 1103. Distribute Candies to People/Solution.py
class Solution:
    def distributeCandies(self, candies: int, num_people: int) -> List[int]:
        lCandies = [0] * num_people
        turn = 1
        counter = 0
        
        while candies:
            if candies >= turn:
                lCandies[counter] += turn
                candies -= turn
                counter += 1
                turn += 1
                counter %= num_people
            else:
                lCandies[counter] += candies
                break
                
        return lCandies
        
